tiledb/sm/cpp_api/query.h

```diff
@@ -1701,16 +1701,13 @@ class Query {
     else
       impl::type_check<char>(schema_.attribute(name).type());
```

Comment:
Since a computation of element_size (previously found) after this is no longer being done,
is this type_check<>() now unnecessary and should also be removed?

Reply:
It's a safety check to ensure the user isn't accidentally using this string-api or a non-string type. Seems OK to be and we perform similar checks in the C++ API.